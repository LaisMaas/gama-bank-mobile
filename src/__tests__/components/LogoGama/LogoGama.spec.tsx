import React from 'react';
import renderer from 'react-test-renderer';
import LogoGama from '../../../components/LogoGama';

jest.mock("@react-navigation/native", () => {
    const actualNav = jest.requireActual("@react-navigation/native")
    return {
      ...actualNav,
      useFocusEffect: () => jest.fn(),
      useNavigation: () => ({
        navigate: jest.fn(),
      }),
    }
  })

jest.mock('react-native-gesture-handler', () => {
  const View = require('react-native/Libraries/Components/View/View');
  return {
    Swipeable: View,
    DrawerLayout: View,
    State: {},
    ScrollView: View,
    Slider: View,
    Switch: View,
    TextInput: View,
    ToolbarAndroid: View,
    ViewPagerAndroid: View,
    DrawerLayoutAndroid: View,
    WebView: View,
    NativeViewGestureHandler: View,
    TapGestureHandler: View,
    FlingGestureHandler: View,
    ForceTouchGestureHandler: View,
    LongPressGestureHandler: View,
    PanGestureHandler: View,
    PinchGestureHandler: View,
    RotationGestureHandler: View,
    /* Buttons */
    RawButton: View,
    BaseButton: View,
    RectButton: View,
    BorderlessButton: View,
    /* Other */
    FlatList: View,
    gestureHandlerRootHOC: jest.fn(),
    Directions: {},
  };
});

describe('<LogoGama />', () => {
  it('Should render <LogoGama />', () => {
    const tree = renderer.create(<LogoGama/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});