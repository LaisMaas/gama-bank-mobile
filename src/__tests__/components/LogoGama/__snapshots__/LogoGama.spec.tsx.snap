// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoGama /> Should render <LogoGama /> 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
        "marginBottom": 0,
        "marginTop": 0,
        "width": 750,
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "maxWidth": 249,
          "width": "100%",
        },
      ]
    }
  />
</View>
`;
